community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/BaseRecordFormats.java

```diff
@@ -123,19 +124,37 @@ public boolean hasCapability( Capability capability )
         return contains( capabilities(), capability );
     }
 
-    public static boolean hasSameCapabilities( RecordFormats one, RecordFormats other, CapabilityType type )
+    public static boolean hasCompatibleCapabilities( RecordFormats one, RecordFormats other, CapabilityType type )
     {
         Set<Capability> myFormatCapabilities = Stream.of( one.capabilities() )
                 .filter( capability -> capability.isType( type ) ).collect( toSet() );
         Set<Capability> otherFormatCapabilities = Stream.of( other.capabilities() )
                 .filter( capability -> capability.isType( type ) ).collect( toSet() );
 
-        return myFormatCapabilities.equals( otherFormatCapabilities );
+        if ( myFormatCapabilities.equals( otherFormatCapabilities ) )
+        {
+            // If they have the same capabilities then of course they are compatible
+            return true;
+        }
+
+        Set<Capability> removedCapabilities =
```

Comment:
i think this is much more shorted equivalent:
```
        boolean capabilitiesNotRemoved = otherFormatCapabilities.containsAll( myFormatCapabilities );

        otherFormatCapabilities.removeAll( myFormatCapabilities );
        boolean allAddedAreAdditive = otherFormatCapabilities.stream().allMatch( Capability::isAdditive );

        // Even if capabilities of the two aren't the same then there's a special case where if the additional
        // capabilities of the other format are all additive then they are also compatible because no data
        // in the existing store needs to be migrated.
        return capabilitiesNotRemoved && allAddedAreAdditive;
```